mapgen/mapgen.cpp

```diff
@@ -0,0 +1,127 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <sstream>
+#include "../Constants.h"
+
+using namespace std;
+
+class Coord {
+	float latitude;
+	float longitude;
+
+  public:
+  	Coord() {
+  		latitude = 0;
+			longitude = 0;
+  	}
+
+		Coord(float latitude, float longitude) {
+			this->latitude = latitude;
+			this->longitude = longitude;
+		}
+
+		Coord(string scoord) {
+			this->longitude = stof(scoord.substr(0, scoord.find(" ")));
+			this->latitude = stof(scoord.substr(scoord.find(" ")));
+		}
+
+	friend std::ostream& operator<< (std::ostream& out, const Coord& z);
+};
+
+ostream& operator<<(ostream& out, const Coord& z){
+	out << "new google.maps.LatLng("<<z.latitude<<","<<z.longitude<<")";
+	return out;
+}
+
+void copyFile(ifstream& input, ofstream& output){
+	string buffer;
+	while (getline(input, buffer)) {
+	    output << buffer << '\n';
+	}
+}
+
+string make_filename( const string& basename, int index, const string& ext )
+{
+  ostringstream result;
+  result << basename << index << ext;
+  return result.str();
+}
+
+void generateMap(int count, vector<Coord>& stores, vector<Coord>& users, vector<Coord>& newStores) {
+
+	if (newStores.size()<count) {
+		cout<<"Error: could not find a new store for the new story query no. "<<count<<endl;
+		return;
+	}
+
+	ifstream part;
+	ofstream map;
+	map.open(make_filename("map", count, ".html"));
+
+	part.open("parts/1.html");
+	copyFile(part, map);
+	part.close();
+
+	for (int i=0; i<stores.size(); i++)
+		map<<stores[i]<<","<<endl;
+
+	part.open("parts/2.html");
+	copyFile(part, map);
+	part.close();
+
+	map<<newStores[count]<<","<<endl;
+
+	part.open("parts/3.html");
+	copyFile(part, map);
+	part.close();
```

Comment:
super cool ideea sa generezi html intreg din bucati si sa umplii variabile de js in c++ :) ; prima oara cand vad asta.